tutorials/plugins/running_code_in_the_editor.rst

```diff
@@ -245,6 +245,147 @@ angle add a setter ``set(new_speed)`` which is executed with the input from the
     to run in the editor too. Autoload nodes cannot be accessed in the editor at
     all.
 
+Getting notified about Resources changing
+-----------------------------------------
+On some occastions you want your tool to use a resource. However, when you change an
+attribute of that resource in the editor, the ``set()`` function of your tool will
+not get called.
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    @tool
+    class_name MyTool
+    extends Node
+
+    @export var resource: MyResource:
+        set(new_resource):
+            resource = new_resource
+            _on_resource_set()
+    
+    # This will only be called, when you create, delete or paste a resource.
+    # You will not get an update when tweaking properteis of it.
+    func _on_resource_set():
+        print("My resource was set!")
+
+ .. code-tab:: csharp
+
+    using Godot;
+
+    [Tool]
+    public partial class MyTool : Node
+    {
+        private float _resource = 1;
+
+        [Export]
+        public MyResource Resource
+        {
+            get => _resource;
+            set
+            {
+                _resource = value;
+                OnResourceSet()
+            }
+        }
+    }
+
+    # This will only be called, when you create, delete or paste a resource.
+    # You will not get an update when tweaking properteis of it
+    private void OnResourceSet()
+    {
+        print("My resource was set!");
+    }
+
+To get around this problem, you first have to make your resource a tool and make it
+emit a signal, whenever a property is set:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    # Make Your Resource a tool.
+    @tool
+    class_name MyResource
+    extends Resource
+
+    # Declare a signal.
+    signal property_updated
```

Comment:
Instead of declaring and emitting a custom signal, we may want to direct users towards the `changed` signal and the `emit_changed` method.

Reply:
Okay, I did not find that signal, it seems really useful.